discovery/discovery_test.go

```diff
@@ -278,9 +728,10 @@ func (tp mockTargetProvider) sendUpdates() {
 
 		time.Sleep(update.interval * time.Millisecond)
 
-		tgs := make([]*config.TargetGroup, len(update.targets))
-		for i, tg := range update.targets {
-			tgs[i] = &config.TargetGroup{Source: tg}
+		tgs := make([]*config.TargetGroup, len(update.targetGroups))
+		for groupIndex := range update.targetGroups {
```

Comment:
Nit: `i` would be more in-line with the Go style guide, see https://github.com/golang/go/wiki/CodeReviewComments#variable-names.